### PR TITLE
Private game notes

### PR DESCRIPTION
**Pull request by turtoise0**
Adds a new chat mode ("personal") in addition to Malkovich, etc. Game notes are only visible to the player.

## Screenshots

Spectator's view:
spectatorview

Player 1 view:
player1view

Player 2 view:
player2view

### CODE REVIEW

**Comment by turtoise0** [comment]
Text-only notes should work, but analysis can't be recorded yet.

File: src/views/Game/GameChat.tsx [hunk 1]
```diff
@@ -761,3 +748,96 @@ function MarkupChatLine({ line }: { line: ChatLine }): JSX.Element {
...
+    );
+}
+
+export function savePersonalGameNote(
+    setPersonalGameNotes,
+    move_number: number,
+    note: string | AnalysisComment | ReviewComment,
+) {
+    const user = data.get("user");
+    setPersonalGameNotes((gameNotes) => {
```
[the remark this answers is not in the document]
Comment (turtoise0) [reply src/views/Game/GameChat.tsx:825]: `setPersonalGameNotes` can also receive an updater function to transform the previous `gameNotes`.

https://reactjs.org/docs/hooks-reference.html#functional-updates

[the remark this answers is not in the document]
Comment (turtoise0) [reply src/views/Game/GameChat.tsx:825]: The type of the update function should be `(game notes: ChatLine[]) => ChatLine[]`

[the remark this answers is not in the document]
Comment (benjaminpjones) [reply src/views/Game/GameChat.tsx:825]: Oh I see, thanks for explaining that :)

> The type of the update function should be `(game notes: ChatLine[]) => ChatLine[]`

Cool, I put that type in (or actually just the React.Dispatch<...> type), and sent you a PR.  It seems to have uncovered inaccuracies in `ChatLine`, which I updated as well.

**Comment by anoek** [comment]
This is quite a clever way to implement this without needing backend changes. Let me look at the backend to gauge how difficult it would be to add there though, it might be pretty straight forward and so I might want to do it through the backend so it's not a separate special code path we have to maintain - it'd also avoid the ever growing sync log that storing private notes in that system would create.

**Comment by anoek** [comment]
Ok, adding proper backend support was pretty straight forward so I did it, but I leave the UI work of figuring out how to nicely deal with log selection, and log coloring, to you.

How it works (and this is on the beta server so you should be able to use it now) is if you send a chat log to the channel "private" it'll send those chats back to you like any other chat when you load the game, so in this way everything else about the system should be the same, including analysis "sharing" etc.

**Comment by anoek** [comment]
I saw you just changed it from "personal" to "private", likely to match what I had - but I had just gone off of the title - now that I see that option though, I think "personal" is a better word to use as "private" might be misconstrued as chat that is only visible between the two players or something. So, sorry for the churn, lets change that back to be "personal" like you had originally :) I'll update the backend in the morning to match.

**Comment by GreenAsJade** [comment]
Nice, this has been wanted for so long!

**Comment by turtoise0** [comment]
see new PR.

**Pull request closed without merging**